plugins/Facebook/class.facebook.plugin.php

```diff
@@ -542,7 +543,7 @@ public function authorizeUri($query = false, $redirectUri = false) {
         }
 
         if ($query) {
-            $redirectUri .= '&'.$query;
+            $redirectUri .= '?'.$query;
```

Comment:
In theory, unless I'm mistaken, someone could still whitelist an URL containing a query string. 
We should detect if $redirectUri contains '?' and in that case add '&' instead of '?'